app/pages/learn/01_tutorial/03_getting-to-know-the-language/03_refactoring_to_functional_style/05_converting_to_streams.md

```diff
@@ -0,0 +1,103 @@
+
+We saw how we can use `filter()` and `map()` functions to select and transform data, respectively. We can perform these operations in the middle of the functional pipeline. In the examples in the previous articles we used functions like `range()` and `rangeClosed()` to create a stream of values in a range of numbers. That worked nicely when we want to iterate over a known range of values, but, often we may want to work with data that comes from external resources, like from a file, for example. If we are able to work with the external resource as a stream, then we can readily apply the operations of the functional pipeline. In this article we'll take a look at an example that illustrates that idea.
+
+<a id="imperativetofunctional">&nbsp;</a>
+## From Imperative to Functional Style
+
+Suppose we want to iterate over a file and count the number of lines with one or more occurrences of a word. Here's an all too familiear imperative style code to accomplish that task:
+
+```java
+//Sample.java
+import java.util.*;
+import java.io.*;
+
+public class Sample {
+  public static void main(String[] args) {
+    try {
+      final var filePath = "./Sample.java";    
+      final var wordOfInterest = "public";
+
+      try (var reader = new BufferedReader(new FileReader(filePath))) {
+        String line = "";
+        long count = 0;
+
+        while((line = reader.readLine()) != null) {
+          if(line.contains(wordOfInterest)) {
+	    count++;
+	  }
+        }
+
+        System.out.println(String.format("Found %d lines with the word %s", count, wordOfInterest));
+      }
+    } catch(Exception ex) {
+      System.out.println("ERROR: " + ex.getMessage());
+    }
+  }
+}
+
+```
+
+In order to make it easy to work with this example, we look for the number of lines with the word "public" in the same source file as the code resides. You may change the value of the `filePath` to refer to a different file and/or the value of the `wordOfInterest` to something else if you like.
+
+
+There are two major parts in this example. We use the `BufferedReader` to access the contents of the file we're interested in looking into. Then, in the `while` loop we check each line to see if it contains the desired word and, if so, increment the `count` to indicate we found another line with the word. Let's examine the two parts, with the second one first.
+
+Looking closely at the loop, from our discussions in the previous articles, we can recognize that the presence of `if` is a sign that we may use the `filter()` operation if we can write the code as a functional pipeline. Once we filter out or select the lines with the desired word, we can count the number of lines, using the `count()` method of stream. You're most likely curious and bursting to ask,  "but, where's the stream?" To answer that question, let's take a look at the first part of the code.
```

Comment:
This line uses two spaces after the comma which is different from other places.
Aside from that, assuming `but, where's the stream?` is a complete question, I think it should start with an uppercase letter.